src/sedtrails/particle_tracer/data_retriever.py

```diff
@@ -1,5 +1,164 @@
+    
+    def get_interpolation_indices(self, target_time: float) -> Tuple[int, int, float]:
+        """
+        Get indices for interpolation between two time steps.
+        
+        Parameters:
+        -----------
+        target_time : float
+            Target time in seconds since reference_date
+            
+        Returns:
+        --------
+        Tuple[int, int, float]
+            (lower_index, upper_index, weight) where weight is the interpolation factor [0-1]
+        """
+        times = self.sedtrails_data.times
+        
+        # Handle edge cases
+        if target_time <= times[0]:
+            return 0, 0, 0.0
+        
+        if target_time >= times[-1]:
+            last_idx = len(times) - 1
+            return last_idx, last_idx, 1.0
+        
+        # Find the index of the last time that is less than or equal to the target time
+        lower_idx = np.searchsorted(times, target_time, side='right') - 1
+        upper_idx = lower_idx + 1
+        
+        # Calculate the interpolation weight
+        time_range = times[upper_idx] - times[lower_idx]
+        
+        # Avoid division by zero
+        if time_range == 0:
+            weight = 0.0
+        else:
+            weight = (target_time - times[lower_idx]) / time_range
+            
+        return lower_idx, upper_idx, weight
+    
+    def get_flow_field(self, time: float) -> Tuple[np.ndarray, np.ndarray]:
+        """
+        Get the flow field at the specified time.
+        
+        This method performs temporal interpolation between the two closest time steps
+        in the SedtrailsData object to obtain the flow field at the requested time.
+        
+        Parameters:
+        -----------
+        time : float
+            Time in seconds since the reference date of the SedtrailsData object
+            
+        Returns:
+        --------
+        Tuple[np.ndarray, np.ndarray]
+            Tuple containing (flow_x, flow_y) arrays
+        """
+        # Get indices for interpolation
+        lower_idx, upper_idx, weight = self.get_interpolation_indices(time)
+        
+        # If time is exactly at a time step or outside the range, no interpolation needed
+        if lower_idx == upper_idx:
+            time_slice = self.sedtrails_data[lower_idx]
+            flow_field = time_slice[self.flow_field_name]
+            return flow_field['x'], flow_field['y']
+        
+        # Otherwise, perform linear interpolation between the two time steps
+        lower_slice = self.sedtrails_data[lower_idx]
+        upper_slice = self.sedtrails_data[upper_idx]
+        
+        lower_flow = lower_slice[self.flow_field_name]
+        upper_flow = upper_slice[self.flow_field_name]
+        
+        # Linear interpolation: result = (1-w)*lower + w*upper
+        flow_x = (1 - weight) * lower_flow['x'] + weight * upper_flow['x']
+        flow_y = (1 - weight) * lower_flow['y'] + weight * upper_flow['y']
+        
```

Comment:
To make this code more modular and testable, I suggest to move the interpolation to its own independent function. If doing it here is not feasible. Could you please open an issue so that we can refactor the code later?